tools/manifest-generator/manifest-generator.go

```diff
@@ -97,44 +189,161 @@ func generateFromFile(templFile string) {
 	}
 	defer file.Close()
 
+	if strings.Contains(*csvVersion, initialCsvVersion) {
+		data.ReplacesCsvVersion = ""
+	} else {
+		klog.Fatalf("Need to implement CSV upgrade to set ReplacesVersion in CSV when CSVVersion is greater than %s\n", initialCsvVersion)
+	}
+
+	data.QuayRepository = *quayRepository
+	data.QuayNamespace = *quayNamespace
+	data.OperatorRules = getOperatorRules()
+	data.OperatorDeploymentSpec = getOperatorDeploymentSpec()
+	data.CDILogo = getCdiLogo(*cdiLogoPath)
+
+	// Read generated manifests in order to populate templated manifest
+	genDir := *genManifestsPath
+
+	data.GeneratedManifests = make(map[string]string)
+
+	manifests, err := ioutil.ReadDir(genDir)
+	if err != nil {
+		klog.Fatalf("Failed to read directory %s: %v\n", templFile, err)
+	}
+
+	for _, manifest := range manifests {
+		if manifest.IsDir() {
+			continue
+		}
+		b, err := ioutil.ReadFile(filepath.Join(genDir, manifest.Name()))
+		if err != nil {
+			klog.Fatalf("Failed to read file %s: %v\n", templFile, err)
+		}
+
+		data.GeneratedManifests[manifest.Name()] = string(b)
+	}
+
 	tmpl := template.Must(template.ParseFiles(templFile))
 	err = tmpl.Execute(os.Stdout, data)
 	if err != nil {
 		klog.Fatalf("Error executing template: %v\n", err)
 	}
 }
 
+func getCdiLogo(path string) string {
+	file, err := os.Open(path)
+	if err != nil {
+		klog.Fatalf("Error retrieving cdi logo file: %v\n", err)
+	}
+
+	// Read entire file into byte slice.
+	reader := bufio.NewReader(file)
+	content, err := ioutil.ReadAll(reader)
+	if err != nil {
+		klog.Fatalf("Error reading cdi logo file: %v\n", err)
+	}
+
+	// Encode as base64.
+	encoded := base64.StdEncoding.EncodeToString(content)
+	return encoded
+}
+
 func generateFromCode(codeGroup string) {
 	var resources []runtime.Object
 
-	crs, err := getClusterResources(codeGroup)
-	if err != nil {
-		klog.Fatalf("Error getting cluster resources: %v\n", err)
+	if isClusterResource(codeGroup) {
+		crs, err := getClusterResources(codeGroup)
+		if err != nil {
+			klog.Fatalf("Error getting cluster resources: %v\n", err)
+		}
+
+		resources = append(resources, crs...)
 	}
 
-	resources = append(resources, crs...)
+	if isOperatorClusterResource(codeGroup) {
```

Comment:
is there any way this function can be more table driven, rather than a bunch of if statements?

Reply:
done

Reply:
done